### PR TITLE
iOS two finger fix

### PR DESCRIPTION
**Pull request by jaisont07**
Added the uitouched to a list and fireevent with the points list to fix the zoom issue in ios

### CODE REVIEW

**Comment by janusw** [comment]
this is supposed to fix issue #10

**Comment by janusw** [comment]
Thanks for the PR, @jaisont07! I just tested it on iOS 17.5 (via the ExampleBrowser in the `OxyplotMauiSample` app, mostly using the example `Annotations -> Tool tips`) and made the following observations ...

Basically two-finger zooming seems to work, but it does not always work perfectly and sometimes seems to behave strangely. E.g. the plot jumps occasionally while doing a two-finger pinch gesture (or when putting down the second finger while the first one is already touching). This behavior seems similar to what I saw on my branch mentioned at https://github.com/oxyplot/oxyplot-maui/issues/10#issuecomment-2227317309.

@jaisont07 Can you reproduce these quirks, and do you have an idea what causes them?

@RobertStefancic Could you maybe also try the branch, and tell us how well it works for you? Does your implementation at https://github.com/oxyplot/oxyplot-maui/issues/10#issuecomment-2227873931 work any better, or does it suffer from the same problems?

The GHA builds actually ran through fine without any problems, which is good. I'll try to look through the code changes soon.

File: Source/OxyPlot.Maui.Skia/Platforms/iOS/Effects/PlatformTouchEffect.cs [hunk 1]
```diff
@@ -88,13 +88,15 @@ public override void TouchesMoved(NSSet touches, UIEvent evt)
...
+        var uitouches = touches.Cast<UITouch>().ToList();
+        foreach (UITouch touch in uitouches)
         {
             long id = ((IntPtr)touch.Handle).ToInt64();
             CheckForBoundaryHop(touch);
             if (idToTouchDictionary[id] != null)
             {
-                FireEvent(idToTouchDictionary[id], id, TouchActionType.Moved, touch, true);
+                //FireEvent with all the points within the touch event
+                FireEvent(idToTouchDictionary[id], id, TouchActionType.Moved, touch, true, uitouches.IndexOf(touch) == uitouches.Count - 1);
```
Comment (janusw) [inline Source/OxyPlot.Maui.Skia/Platforms/iOS/Effects/PlatformTouchEffect.cs:99]: I guess the same multi-point handling that you're using here should also be applied to the `FireEvent` calls in `TouchesBegan` and `TouchesEnded` ?

**Comment by janusw** [comment]
@jaisont07 @RobertStefancic Ping! Is there still interest in this feature? If yes, could you reply to my comments, so that we can move forward with this? I think it would be great to add proper zoom support for iOS ...

**Comment by jaison-t-john** [comment]
> @jaisont07 @RobertStefancic Ping! Is there still interest in this feature? If yes, could you reply to my comments, so that we can move forward with this? I think it would be great to add proper zoom support for iOS ...

@janusw Yes this is required. But I am also able to reproduce the not so smooth zoom issue with all these fixes. Would be great if we could find a solution to this..

**Comment by RobertStefancic** [comment]
@janusw @jaisont07 I created a pull-request as well here: https://github.com/oxyplot/oxyplot-maui/pull/15
This is the code that we used for the public release and we haven't had any complaints so far. The intention is to mimic how the Android touch events are dispatched, because on there it works fine.

Edit: First thing I noticed for the snappy behavior is that the touch events where coming rarely (which is different from Android, where they come often), hence I added the ShouldRecognizeSimultaneously => true.
Afterwards, I had to synchronize how many fingers are pressed at once - that is why I made it to only trigger the actual move event when the user has all the fingers in place. Otherwise, it will again appear snappy because the user touches will trigger at different locations and will move the zoom on the chart.

**Comment by janusw** [comment]
I'm closing this PR in favor of #15 (which was just merged).

**Pull request closed without merging**